src/frontend/screens/Settings/components/Gamescope.tsx

```diff
@@ -0,0 +1,334 @@
+    return undefined
+  }
+
+  if (!isLinux) {
+    return <></>
+  }
+
+  if (fetching) {
+    return (
+      <UpdateComponent
+        message={t(
+          'settings.gamescope.searchMsg',
+          'Searching for gamescope executable.'
+        )}
+      />
+    )
+  }
+
+  if (!isInstalled) {
+    return (
+      <div style={{ color: 'red' }}>
+        {t(
+          'setting.gamescope.missingMsg',
+          'We could not found gamescope on the PATH. Install it or add it to the PATH.'
+        )}
+      </div>
+    )
+  }
+
+  return (
+    <div>
+      {/* Enable Upscale */}
+      <div className="toggleRow">
+        <ToggleSwitch
+          htmlId="gamescopeUpscaleToggle"
+          value={gamescope.enableUpscaling || false}
+          handleChange={() =>
+            setGamescope({
+              ...gamescope,
+              enableUpscaling: !gamescope.enableUpscaling
+            })
+          }
+          title={t('setting.gamescope.enableUpscaling', 'Enables Upscaling')}
+        />
+      </div>
+      {/* Upscale Settings */}
+      {gamescope.enableUpscaling && (
+        <>
+          {/* Upscale Method */}
+          <SelectField
+            label={'Upscale Method'}
+            htmlId="upscaleMethod"
+            afterSelect={
+              <FontAwesomeIcon
+                className="helpIcon"
+                icon={faCircleInfo}
+                title={t(
+                  'help.gamescope.upscaleMethod',
+                  'The upscaling method gamescope should use.'
+                )}
+              />
+            }
+            onChange={(event: ChangeEvent<HTMLSelectElement>) =>
+              setGamescope({
+                ...gamescope,
+                upscaleMethod: event.currentTarget.value
+              })
+            }
+            value={gamescope.upscaleMethod}
+          >
+            {['fsr', 'nis', 'integer', 'stretch'].map((opt, i) => (
+              <option key={i}>{opt}</option>
+            ))}
+          </SelectField>
+          {/* Game Res */}
+          <div className="toggleRow">
+            <TextInputField
+              label={t('options.gamescope.gameWidth', 'Game Width')}
+              htmlId="gameWidth"
+              placeholder=""
+              maxLength={4}
+              value={gamescope.gameWidth}
+              afterInput={
+                <FontAwesomeIcon
+                  className="helpIcon"
+                  icon={faCircleInfo}
+                  title={t(
+                    'help.gamescope.gameWidth',
+                    'The width resolution used by the game. A 16:9 aspect ratio is assumed by gamescope.'
+                  )}
+                />
+              }
+              onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                setGamescope({
+                  ...gamescope,
+                  gameWidth:
+                    setResolution(event.currentTarget.value) ??
+                    gamescope.gameWidth
+                })
+              }}
```

Comment:
Just a suggestion, we use this code in 4 places, maybe we could move this inside `setResolution` and do `setResolution('gameWidth', event.currentTarget.value)` for example

and do this `setGamescope....` code there instead